pkg/cli/admin/top/persistentvolumeclaims.go

```diff
@@ -0,0 +1,385 @@
+package top
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+	"k8s.io/klog/v2"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+)
+
+const (
+	localhostRecoveryTokenSecret = "localhost-recovery-client-token"
+	kubeApiserverNamespace       = "openshift-kube-apiserver"
+)
+
+var (
+	topPersistentVolumeClaimsLong = templates.LongDesc(`
+		Experimental: Show usage statistics for bound persistentvolumeclaims.
+
+		This command analyzes all the bound persistentvolumeclaims managed by the platform and presents current usage statistics.
+	`)
+
+	topPersistentVolumeClaimsExample = templates.Examples(`
+		# Show usage statistics for all the bound persistentvolumeclaims across the cluster
+		oc adm top persistentvolumeclaims -A
+
+		# Show usage statistics for all the bound persistentvolumeclaims in a specific namespace
+		oc adm top persistentvolumeclaims -n default
+
+		# Show usage statistics for specific bound persistentvolumeclaims 
+		oc adm top persistentvolumeclaims database-pvc app-pvc -n default
+
+	`)
+)
+
+// RouteGetter is a function that gets a Route.
+type RouteGetter func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error)
+
+type options struct {
```

Comment:
`options` is unexported which makes sense but why some of its fields are exported?

Reply:
fixed